src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,364 @@
+---
+title: "Codelab: basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad2 code examples."
+toc: true
+--- 
+{{site.alert.important}}
+  This codelab covers basic Flutter layout concepts using a code editor called DartPad2. An understanding of [Dart](https://www.dart.dev) or some exposure to object-oriented programming is recommended. 
+{{site.alert.end}}
+{{site.alert.important}}
+  Currently, DartPad2 is in the experimental stage and being tested with Google Chrome. A PC (Microsoft, Mac, or Linux) with Google Chrome installed is required for this codelab. If you encounter any difficulties while using DartPad2, please create a [DartPad issue](https://github.com/dart-lang/dart-pad/issues/new) and label the issue "platform-web."
+{{site.alert.end}} 
+{{site.alert.note}}
+  Flutter supports devices that maintain right-to-left language settings. 
+{{site.alert.end}}
+## Introduction
+In this codelab, you interact with and build Flutter layouts in a code editor called DartPad2. As you progress through this codelab, you learn that widgets build everything in flutter. A widget is an immutable object that describes a specific part of a UI. You also learn that, while widgets are simple in their functions, widgets can be complex in their structures. Widgets can be built around and inside of other widgets, and widgets can have interconnected and ordered relationships with one another. At the end of this codelab, you apply what you learn and build a simple interface using basic Fluter layout concepts. 
+## Row class and Column class
+`Row` and `Column` are classes that contain widgets. When `Row` and `Column` contain widgets, the widgets become "children," and `Row` and `Column` become "parents." Moreover, when `Row` and `Column` contain widgets, `Row` and `Column` lay them out. `Row` lays out its widgets horizontally. `Column` lays out its widgets vertically. 
+#### Code example: Row
+{:.no_toc}
+{{site.alert.secondary}}
+{:.no_toc}
+  *Check out Dart.*
+
+  In the code, `Row` contains three children, all `BlueBox` widgets. 
+
+  In this code example, the code displays three blue squares horizontally.
+
+  ***Run the code.***
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/26c8fb02c58c4120917b073ec6253953
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=26c8fb02c58c4120917b073ec6253953&amp;fw=true&amp;split=60" width="100%" height="400px"></iframe> 
+
+#### Code example: Column
+{:.no_toc}
+{{site.alert.secondary}}
+{:.no_toc}
+  *Check out Dart.*
+
+  In the code, `Column` contains three child widgets, all `BlueBox` widgets. 
+
+  In this code example, the code displays three blue squares vertically.
+
+  ***Run the code.***
```

Comment:
I'd probably say "Click the Run button" but, again, let's get consensus with Kathy.

Reply:
Agreed. I'm usually saying "click the **Run** button".

Reply:
Okay, will do.